server/src/main/java/io/littlehorse/server/streamsimpl/BackendInternalComms.java

```diff
@@ -858,22 +858,28 @@ private InternalScanReplyPb specificPartitionTagScan(InternalScan search) {
             );
 
             String prefix = search.getTagScan().getKeyPrefix() + "/";
-            List<ByteString> result = store
-                .prefixTagScanStream(prefix, Tag.class)
-                .limit(search.limit)
-                .map(tag -> {
-                    var objectId = tag.getDescribedObjectId();
-                    return ObjectId
-                        .fromString(
-                            objectId,
-                            Getable.getIdCls(search.getObjectType())
-                        )
-                        .toProto()
-                        .build()
-                        .toByteString();
-                })
-                .collect(Collectors.toList());
-            out.addAllResults(result);
+            LHKeyValueIterator<Tag> tagScanResultIterator = store.prefixScan(
```

Comment:
do you call `tagScanResultIterator.close()`? If you don't, then we leak resources on the underlying RocksDB process, which is dangerous.

I think the TagScanStream thing didn't close it either...